pkg/dsl/definition/template.go

```diff
@@ -35,53 +38,43 @@ const (
 	AuxiliaryWorkload = "AuxiliaryWorkload"
 )
 
-var (
-	metadataAccessor = meta.NewAccessor()
-)
-
-// Template defines Definition's Render interface
-type Template interface {
-	Params(params interface{}) Template
-	Complete(ctx process.Context) error
-	Output(ctx process.Context, client client.Client, name string) Template
-	HealthCheck() error
+// AbstractEngine defines Definition's Render interface
+type AbstractEngine interface {
+	Params(params interface{}) AbstractEngine
+	Complete(ctx process.Context, abstractTemplate string) error
```

Comment:
normally, we complete something given the "parameter" instead of the template. Am I missing something?